contrib/pg_tde/src/access/pg_tde_xlog_smgr.c

```diff
@@ -166,7 +185,8 @@ TDEXLogShmemInit(void)
 
 typedef struct EncryptionStateData
 {
-	XLogRecPtr	enc_key_lsn;	/* to sync with reader */
+	XLogRecPtr	enc_key_tli;
```

Comment:
I know this is not new, but this double EncryptionStateData seems strage to me - postgres doesn't have support for the atomic types in the frontend, or why do we have to duplicate them?

Also, now we also have a sizedifference, tli is 32 bit in the backend code, and 64 in the frontend.

Reply:
1. Yes, frontend doesn't support atomics:
```c
...
#define ATOMICS_H

#ifdef FRONTEND
#error "atomics.h may not be included from frontend code"
#endif
...
```

2. My bad, it must be a `TimeLineID` type. Fixed.